components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -83,17 +84,23 @@ export default function () {
         })();
     }, [project, team, teams]);
 
+    const updateProjectPartial = (settings: DeepPartial<ProjectSettings | undefined>) => {
+        if (!project) {
+            return;
+        }
+
+        return getGitpodService().server.updateProjectPartial({ id: project.id, settings });
+    };
```

Comment:
The partial updates 'worked' with a single value specified, but the value not specified would be set to its default of `false`.

The wrapper function is not necessary but just cleans up the two handler functions by saving a line or two of duplicated code.

Reply:
Thanks for explaining!

> but the value not specified would be set to its default of `false`.

Oh. This sounds like a bug -- shouldn't it rather stay at its previously set value?

Reply:
Ah, or maybe it's a race condition if you update both values really fast? 🤔 (I.e. the second DB save would still have the old value because the first DB save didn't happen fast enough?)

Reply:
Yes, the semantics of a partial update should leave unspecified values unchanged. Not sure why that isn't happening here and tbh didn't investigate beyond what was necessary to fix this UI glitch.

Reply:
I think the ideal solution would be to implement a lock / DB transaction (i.e. only one partial update can run at any time, others get queued and run after the lock is freed??)